packages/frontend/core/src/components/blocksuite/block-suite-header/menu/index.tsx

```diff
@@ -148,16 +149,22 @@ export const PageHeaderMenuButton = ({
   }, [rename]);
 
   const handleSwitchMode = useCallback(() => {
-    editorService.editor.toggleMode();
+    const mode = primaryMode === 'page' ? 'edgeless' : 'page';
+    editorService.editor.doc.setPrimaryMode(mode);
```

Comment:
updating primary mode only happens in page property settings

Reply:
But changing the primary mode in the menu is a requirement